x-pack/solutions/observability/plugins/synthetics/server/tasks/sync_private_locations_monitors_task.ts

```diff
@@ -119,27 +121,33 @@ export class SyncPrivateLocationMonitorsTask {
         taskState,
       });
 
-      if (hasDataChanged || performSync) {
-        if (hasDataChanged) {
+      // Only perform syncGlobalParams if:
+      // - hasDataChanged and disableAutoSync is false
+      // - OR performSync is true (from cleanup), regardless of disableAutoSync
+      if ((hasDataChanged && !taskState.disableAutoSync) || performSync) {
+        if (hasDataChanged && !taskState.disableAutoSync) {
           this.debugLog(`Syncing private location monitors because data has changed`);
-        } else {
+        } else if (performSync) {
```

Comment:
This is more explicit but not needed. There are two ways into this block and if the first one is `false` this one is always truthy. But perhaps best to leave as is to insulate against future changes modifying the behavior?